cpp/CMakeLists.txt

```diff
@@ -149,47 +135,67 @@ set(RAFT_LINK_LIBRARIES
         rmm::rmm
         cuco::cuco
         )
-
 target_link_libraries(raft INTERFACE ${RAFT_LINK_LIBRARIES})
-target_link_libraries(raft_distance PUBLIC ${RAFT_LINK_LIBRARIES})
-target_link_libraries(raft_nn PUBLIC ${RAFT_LINK_LIBRARIES} FAISS::FAISS)
 
 set(RAFT_COMPILE_DEFINITIONS
         $<$<BOOL:${NVTX}>:NVTX_ENABLED>
         )
-
 target_compile_definitions(raft INTERFACE ${RAFT_COMPILE_DEFINITIONS})
-target_compile_definitions(raft_distance PRIVATE ${RAFT_COMPILE_DEFINITIONS})
-target_compile_definitions(raft_nn PRIVATE ${RAFT_COMPILE_DEFINITIONS})
-
-target_compile_options(raft_distance
```

Comment:
When I add these back, the final packages seem to compile successfully. Are these required in general?